File: eventbus/app/src/main/java/com/example/eventbus/BlankFragment.java
package com.example.eventbus;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.eventbus.library.BlankBaseLibraryFragment;

import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

public class BlankFragment extends BlankBaseLibraryFragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_blank, container, false);
    }

    @Override
    @Subscribe(threadMode = ThreadMode.MAIN_ORDERED)
    // subscribe annotation in base class would not be picked up by index
    public void handleEvent(BlankBaseLibraryFragment.SampleEvent event) {
        super.handleEvent(event);
    }
}

File: eventbus-kotlin/app/src/main/java/com/example/eventbuskotlin/BlankFragment.kt
package com.example.eventbuskotlin

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import org.greenrobot.eventbus.Subscribe

class BlankFragment : BlankBaseFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_blank, container, false)
    }

    @Subscribe // subscribe annotation in base class would not be picked up by index
    override fun handleEvent(event: BlankBaseFragment.SampleEvent) {
        super.handleEvent(event)
    }

    class SampleEvent
}
